File: 822/solution.go
package _822

func flipgame(fronts []int, backs []int) int {
	same := map[int]struct{}{}
	keys := map[int]struct{}{}
	for i, v1 := range fronts {
		v2 := backs[i]
		if v1 == v2 {
			same[v1] = struct{}{}
		} else {
			keys[v1] = struct{}{}
			keys[v2] = struct{}{}
		}
	}
	res := 2001
	min := func(a, b int) int {
		if a < b {
			return a
		}
		return b
	}
	for k, _ := range keys {
		_, ok := same[k]
		if !ok {
			res = min(res, k)
		}
	}
	if res == 2001 {
		res = 0
	}
	return res
}

File: 822/solution.py
from typing import List


class Solution:
    def flipgame(self, fronts: List[int], backs: List[int]) -> int:
        n = len(fronts)
        keys = set()
        not_allow = set()
        for i in range(n):
            keys.add(fronts[i])
            keys.add(backs[i])
            if fronts[i] == backs[i]:
                not_allow.add(fronts[i])
        res = 2001
        for k in keys:
            if k not in not_allow:
                res = min(res, k)
        return 0 if res == 2001 else res
